### ISSUE TITLE
Proposal: Treat goog.scope as IIFE in indent rule

### ISSUE THREAD

**Issue by jschaf** [eslint/eslint#7190]
I'd like to modify the indent rule to treat code within a `goog.scope` function expression similarly to an IIFE.  The main reason is to correctly lint code that follows Google's style guide.

I'm willing to implement this change.

For example, the following has a valid indent according to the google style guide:

```js
goog.scope(function() {
var Button = goog.ui.Button;
var dom = goog.dom;

SomeType.prototype.findButton = function() {
  var myText = "hi";
};
});  // goog.scope
```

Previous issues about this topic:

- https://github.com/eslint/eslint/issues/6259
- https://github.com/eslint/eslint/issues/6585
- https://github.com/eslint/eslint/issues/6613

**Comment by ljharb** [comment]
@jschaf https://github.com/eslint/eslint/issues/6259 implies that there's an `outerIIFEBody` option to the rule for precisely this use case. Does setting that not resolve your issue?

**Comment by platinumazure** [comment]
I know this is covered by another issue, but I can't search for it easily on the mobile interface.

**Comment by ljharb** [comment]
ah, i see https://github.com/eslint/eslint/issues/6585#issuecomment-230361533 now

**Comment by ljharb** [comment]
so yes, that's not an IIFE - that seems like something that a separate google-specific rule (not in core) should probably address.

**Comment by jschaf** [comment]
The problem is that a plugin for this case would require re-implementing most of the indent functionality.  I figured the use case was common enough to include in eslint

**Comment by vitorbal** [comment]
@jschaf Thank you for your issue. Could you do us a big favor and fill in the rest of the [template for proposing a rule change](https://github.com/eslint/eslint/blob/master/templates/rule-change-proposal.md)?

Also, in my opinion, for this enhancement to have a chance of being accepted in core, the rule option should be something that can be formalized in a more "generic" way and not specific to one framework and/or style guide.
Otherwise, as @ljharb mentioned, your best bet would be creating a separate google-specific custom rule or plugin.

**Comment by platinumazure** [comment]
Sorry for taking so long to find this. [This comment](https://github.com/eslint/eslint/issues/6690#issuecomment-233412041) covers a proposal we are entertaining which would fix this issue. So I'd like to close as a duplicate if the team doesn't object. Thanks!

**Comment by vitorbal** [comment]
Sounds good to me!
